src/main/java/com/gizasystems/automationexercise/apis/ApisSearchProducts.java

```diff
@@ -0,0 +1,46 @@
+package com.gizasystems.automationexercise.apis;
+
+import com.gizasystems.automationexercise.utils.HtmlResponse;
+import com.shaft.api.RestActions;
+import com.shaft.driver.SHAFT;
+import io.qameta.allure.Step;
+import io.restassured.http.ContentType;
+import java.util.Arrays;
+import java.util.List;
+
+
+public class ApisSearchProducts {
+    private SHAFT.API api;
+    public ApisSearchProducts(SHAFT.API api) {
+        this.api = api;
+    }
+    // Services
+    private static final String searchProduct_serviceName = "/searchProduct";
+
+
+    @Step("API Search For Product")
+    public ApisSearchProducts searchProductApi(String searchProductFiled, String searchedProduct) {
+        List<List<Object>> formData = Arrays.asList(
+                Arrays.asList(searchProductFiled, searchedProduct)
+
+        );
+        api.post(searchProduct_serviceName)
+                .setParameters(formData, RestActions.ParametersType.FORM)
+                .setContentType(ContentType.URLENC)
+                .setTargetStatusCode(Apis.SUCCESS)
+                .perform();
+        return this;
+    }
+
+    //////////////////// Validations \\\\\\\\\\\\\\\\\\\\
+    public ApisSearchProducts validateOnCategory() {
```

Comment:
the category should be in the method parameters to by dynamic when used for other categories by other users.